### ISSUE TITLE
Search results for binary (TANIMOTO) is wrong

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#6569]
#### Describe the bug:
Search results for binary (TANIMOTO) is wrong when both auto_id and _async are true
collection.num_entities
2
len(res[0])
1
#### Expected behavior:
 len(res[0]) == nb when limit > nb

#### Steps/Code to reproduce:
```python
from pymilvus_orm import CollectionSchema, FieldSchema
from pymilvus_orm import Collection
from pymilvus_orm import connections
from pymilvus_orm.types import DataType
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
binary_vector = FieldSchema(name="binary_vector", dtype=DataType.BINARY_VECTOR, dim=128)
schema = CollectionSchema(fields=[int64_field, float_field, binary_vector], auto_id=True)
connections.connect()
collection = Collection("test_search_collection_binbin_8", schema=schema)
dim = 128
nb = 2
nq = 2
limit = 10
import random
raw_vectors = []
binary_vectors = []
for _ in range(nb):
    raw_vector = [random.randint(0, 1) for _ in range(dim)]
    raw_vectors.append(raw_vector)
    binary_vectors.append(bytes(np.packbits(raw_vector, axis=-1).tolist()))
res = collection.insert([[np.float32(i) for i in range(nb)], binary_vectors])
collection.num_entities
collection.load()
default_index = {"index_type": "BIN_IVF_FLAT", "params": {"nlist": 128}, "metric_type": "TANIMOTO"}
collection.create_index("binary_vector", default_index)
search_params = {"metric_type": "TANIMOTO", "params": {"nprobe": 10}}
res1 = collection.search(binary_vectors[:nq], "binary_vector", search_params, limit, "int64 >= 0", _async=True)
res1.done()
res = res1.result()
len(res)
len(res[0])
```

Results:
```
collection.num_entities
2
res1 = collection.search(binary_vectors[:nq], "binary_vector", search_params, limit, "int64 >= 0", _async=True)
res1.done()
res = res1.result()
len(res)
2
len(res[0])
1
```

#### Environment:
- Milvus version(e.g. v1.1.0 or 8b23a93): latest
- SDK version(e.g. pymilvus v2.0.0rc2):
Name: pymilvus
Version: 2.0.0rc3.dev4
Name: pymilvus-orm
Version: 2.0.0rc3.dev5

- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:

**Comment by DragonDriver** [comment]
How about other metric type? For example, JACCARD?

**Comment by DragonDriver** [comment]
I don't think it's `_async` caused this issue. I reproduced this even when the `_async` is False. Any way, I will fix this soon and thx for pointing out this.

**Comment by binbinlv** [comment]
> How about other metric type? For example, JACCARD?

Others are good.

**Comment by binbinlv** [comment]
> I don't think it's `_async` caused this issue. I reproduced this even when the `_async` is False. Any way, I will fix this soon and thx for pointing out this.

OK, thanks.

**Comment by DragonDriver** [comment]
this will be solved by #6628 , pls retest this after it was merged. thx

/unassign
/assign @binbinlv

**Comment by binbinlv** [comment]
The problem still exists:

===================================================================================== short test summary info ======================================================================================
FAILED test_search.py::TestCollectionSearch::test_search_binary_tanimoto_flat_index[500-8-False-False] - AssertionError
FAILED test_search.py::TestCollectionSearch::test_search_binary_tanimoto_flat_index[500-8-False-True] - AssertionError
FAILED test_search.py::TestCollectionSearch::test_search_binary_tanimoto_flat_index[500-8-True-False] - AssertionError
FAILED test_search.py::TestCollectionSearch::test_search_binary_tanimoto_flat_index[500-8-True-True] - AssertionError

[2021-07-21 03:07:14 - ERROR - ci_test]: search_results_check: limit(topK) searched (1) is not equal with expected (2) (func_check.py:186)
[2021-07-21 03:07:21 - ERROR - ci_test]: search_results_check: limit(topK) searched (1) is not equal with expected (2) (func_check.py:186)
[2021-07-21 03:07:27 - ERROR - ci_test]: search_results_check: limit(topK) searched (1) is not equal with expected (2) (func_check.py:186)
[2021-07-21 03:07:33 - ERROR - ci_test]: search_results_check: limit(topK) searched (0) is not equal with expected (2) (func_check.py:186)

@DragonDriver

**Comment by binbinlv** [comment]
Log is uploaded as the following if it is a help:
[log.tar.zip](https://github.com/milvus-io/milvus/files/6854989/log.tar.zip)

**Comment by DragonDriver** [comment]
It cannot be reproduced on my local pc now, can you please retry this? thx @binbinlv

**Comment by binbinlv** [comment]
@DragonDriver OK, will retry it later.

**Comment by yanliang567** [comment]
@binbinlv any chance to verify it before rc3

**Comment by binbinlv** [comment]
Yes, tt is not fixed yet. 

milvus(master): 61786f7
Name: pymilvus
Version: 2.0.0rc3.dev6
Name: pymilvus-orm
Version: 2.0.0rc3.dev9

[2021-07-29 06:38:38 - ERROR - ci_test]: search_results_check: limit(topK) searched (1) is not equal with expected (2) (func_check.py:186)

FAILED test_search.py::TestCollectionSearch::test_search_binary_tanimoto_flat_index[2-8-True-True]

**Comment by yanliang567** [comment]
get off from rc3 as binary vector is not a common case

**Comment by cydrain** [comment]
fixed by #6905

**Comment by binbinlv** [comment]
Verified and fixed.

milvus(master): 87c5a49
Name: pymilvus
Version: 2.0.0rc3.dev16
Name: pymilvus-orm
Version: 2.0.0rc3.dev15
